lib/src/code_field/code_controller.dart

```diff
@@ -334,6 +334,151 @@ class CodeController extends TextEditingController {
     );
   }
 
+  void outdentSelection() {
+    final tabSpaces = params.tabSpaces;
+    if (selection.start == -1 || selection.end == -1) {
+      return;
+    }
+
+    modifySelectedLines(
+      (line) {
+        if (line == '\n') {
+          return line;
+        }
+
+        if (line.length < tabSpaces) {
+          return line.trimLeft();
+        }
+
+        final subStr = line.substring(0, tabSpaces);
+        if (subStr == ' ' * tabSpaces) {
+          return line.substring(tabSpaces, line.length);
+        }
+        return line.trimLeft();
+      },
+      shouldSustainSelection: true,
+    );
+  }
+
+  void indentSelection() {
+    final tabSpaces = params.tabSpaces;
+    final tab = ' ' * tabSpaces;
+    final lines = _code.lines.lines;
+    if (selection.start == -1 || selection.end == -1) {
+      return;
+    }
+
+    if (selection.isCollapsed) {
+      final cursor = _code.hiddenRanges.recoverPosition(
+        selection.start,
+        placeHiddenRanges: TextAffinity.downstream,
+      );
+      final lineIndex = _code.lines.characterIndexToLineIndex(cursor);
+      final columnIndex = cursor - lines[lineIndex].textRange.start;
+      final insert = ' ' * (tabSpaces - (columnIndex % tabSpaces).abs());
```

Comment:
What is the case when `.abs()` is required?

Reply:
I misplaced it, but it is not required anyway xD